staging/src/k8s.io/component-base/config/options/leaderelectionconfig.go

```diff
@@ -40,10 +40,9 @@ func BindLeaderElectionFlags(l *config.LeaderElectionConfiguration, fs *pflag.Fl
 	fs.DurationVar(&l.RetryPeriod.Duration, "leader-elect-retry-period", l.RetryPeriod.Duration, ""+
 		"The duration the clients should wait between attempting acquisition and renewal "+
 		"of a leadership. This is only applicable if leader election is enabled.")
-	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
-		"The type of resource object that is used for locking during "+
-		"leader election. Supported options are 'leases', 'endpointsleases' "+
-		"and 'configmapsleases'.")
+	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", "leases", ""+
+		"Deprecated: The type of resource object that is used for locking during "+
+		"leader election. This option setting will not take effect, Options locked to 'leases'")
```

Comment:
revert this change or make it conditional on a parameter that kube-scheduler/kube-controller-manager, etc can pass in. We will continue supporting endpointsleases and configmapsleases for out of tree callers.